client/app/components/Toast/index.jsx

```diff
@@ -0,0 +1,73 @@
+// @flow
+import React, { useState } from 'react';
+import type { Node } from 'react';
+import { I18n } from 'libs/i18n';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import css from './Toast.scss';
+
+type Props = {
+  alert?: string,
+  notice?: string,
+  appendDashboardClass?: boolean
+};
+export type State = {
+  showToast: boolean,
+};
+
+export const Toast = ({
+  alert, notice, appendDashboardClass,
+}: Props): Node => {
+  const [showAlert, setShowAlert] = useState<boolean>(alert !== null && alert !== '' && !document.documentElement?.hasAttribute('data-turbolinks-preview'));
+  const [showNotice, setShowNotice] = useState<boolean>(notice !== null && notice !== '' && !document.documentElement?.hasAttribute('data-turbolinks-preview'));
+  const hideNotice = () => {
+    setShowNotice(false);
+  };
+  const hideAlert = () => {
+    setShowAlert(false);
+  };
+  if (showAlert || showNotice) {
+    setTimeout(() => {
+      hideNotice();
+      hideAlert();
+    }, 7000);
+  }
+  return (
+    <div>
+      <div id="toast-notice" aria-label={showNotice ? I18n.t('alert_auto_hide') : ''} style={{ visibility: showNotice ? 'visible' : 'hidden' }} role="region" aria-live="polite" aria-atomic="true" className={`${showNotice ? 'notice' : ''} ${css.toast} ${showNotice && (showAlert || appendDashboardClass) ? 'smallMarginBottom' : ''}`}>
```

Comment:
I think you'll want to only render this entire element only if `showNotice` exists.